### PR TITLE
Update Azure deps and async impl

### PR DESCRIPTION
**Pull request by jborean93**
Updates the versions of the Azure deps to the latest available today. Also updates the AsyncPSCmdlet implementation with a better tested variant that fixes up some edge race conditions and utilitises a saner implementation.

### CODE REVIEW

File: src/OpenAuthenticode/Commands/OpenAuthenticodeAzTrustedSigner.cs [hunk 1]
```diff
@@ -56,15 +57,15 @@ protected override async Task ProcessRecordAsync()
...
             Response<Stream> chainResponse = await client.GetSignCertificateChainAsync(
                 codeSigningAccountName: AccountName,
                 certificateProfileName: ProfileName,
-                cancellationToken: CancelToken).ConfigureAwait(false);
+                cancellationToken: cancellationToken).ConfigureAwait(false);
 
             rawChain = new byte[chainResponse.Value.Length];
             await chainResponse.Value.CopyToAsync(
-                new MemoryStream(rawChain)).ConfigureAwait(false);
+                new MemoryStream(rawChain), cancellationToken).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (jborean93) [reply src/OpenAuthenticode/Commands/OpenAuthenticodeAzTrustedSigner.cs:68]: It doesn't throw and works as is today.
